thewalrus/grouped_click_probabilities.py

```diff
@@ -0,0 +1,73 @@
+"""Functions for computing grouped click probabilities"""
+import numpy as np
+from numba import jit
+
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-locals
+@jit(nopython=True)
+def grouped_click_probabilities(phn, chn, t_matrix, num_samples, num_groups, seed=1990):
+    """Computes click probabilities and errors for input states sent into a lossy interferometer
+
+    Args:
+        phn (array): mean photon numbers of input modes
+        chn (array): coherences of input modes
+        t_matrix (array): transfer matrix
+        num_samples (int): number of samples
+        num_groups (int): number of groups into which the samples are divided for error computation
+        seed (int): seed of the random number generator
+
+    Returns:
+        tuple (prob, error): array of grouped click probabilities and array of corresponding errors
+
+    """
+    np.random.seed(seed)
+    num_modes, num_input = max(t_matrix.shape), min(t_matrix.shape)
+    drp = np.array([(0.5 * np.complex128(phn[i] + chn[i])) ** 0.5 for i in range(num_input)])
+    drm = np.array([(0.5 * np.complex128(phn[i] - chn[i])) ** 0.5 for i in range(num_input)])
+    delta = 2 * np.pi / (num_modes + 1)
+    f_mat = np.asarray(
+        [
+            [(1 / (num_modes + 1)) * np.exp(1j * q * p * delta) for q in range(num_modes + 1)]
+            for p in range(num_modes + 1)
+        ]
+    )
+    acc = np.zeros(num_modes + 1, dtype=np.float64)
+    bcc = np.zeros(num_modes + 1, dtype=np.float64)
+    qcc = np.zeros(num_modes + 1, dtype=np.float64)
+    fix = np.zeros(num_modes + 1, dtype=np.float64)
+    for j in range(num_samples):
+        wrp = np.array([np.random.normal() for _ in range(num_input)])
+        wrm = np.array([np.random.normal() for _ in range(num_input)])
```

Comment:
Once using a local instance of the RNG, this two lines can be modified to use it. Also, you can use the `size` kwarg to get an array of values drawn from the distribution.
```suggestion
        wrp = rng.normal(size=(num_input,))
        wrm = rng.normal(size=(num_input,))
```

Reply:
Then, you can dismiss this. Marking it as resolved.